packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -134,22 +134,61 @@ const PipelineWrapper: React.FC<IProps> = ({
     const currentContext = contextRef.current;
 
     const changeHandler = (): void => {
-      const pipeline = currentContext.model.toJSON();
-      setPipeline(pipeline);
+      const pipelineJson: any = currentContext.model.toJSON();
+      if (pipelineJson?.pipelines?.[0]?.nodes) {
+        // Update to display actual value of runtime image
+        for (const node of pipelineJson?.pipelines?.[0]?.nodes) {
+          const app_data = node?.app_data;
+          if (app_data?.runtime_image) {
+            if (runtimeImages.current?.[app_data.runtime_image]) {
+              app_data.runtime_image =
+                runtimeImages.current?.[app_data.runtime_image];
+            }
+          }
+        }
+      }
+      setPipeline(pipelineJson);
       setLoading(false);
     };
 
     currentContext.model.contentChanged.connect(changeHandler);
 
     currentContext.ready.then(changeHandler);
 
+    PipelineService.getRuntimeImages()
+      .then((images: any) => {
+        runtimeImages.current = images;
+        const nodesCopy = JSON.parse(JSON.stringify(nodes));
+        for (const node of nodesCopy) {
+          node.properties.uihints.parameter_info[1].data.items = Object.values(
+            runtimeImages.current
+          );
+        }
+        setUpdatedNodes(nodesCopy);
+        changeHandler();
+      })
+      .catch(error => RequestErrors.serverError(error));
+
     return (): void => {
       currentContext.model.contentChanged.disconnect(changeHandler);
     };
   }, []);
 
   const onChange = useCallback((pipelineJson: any): void => {
     if (contextRef.current.isReady) {
+      if (pipelineJson?.pipelines?.[0]?.nodes) {
+        // Update to store tag of runtime image
+        for (const node of pipelineJson?.pipelines?.[0]?.nodes) {
+          const app_data = node?.app_data;
+          if (app_data?.runtime_image) {
+            for (const tag in runtimeImages.current) {
+              if (runtimeImages.current?.[tag] === app_data?.runtime_image) {
+                app_data.runtime_image = tag;
+              }
+            }
+          }
+        }
+      }
```

Comment:
I'm a little hesitant about directly mutating the pipeline json here, but if it's not breaking anything 🤷‍♂️